apps/web-giddh/src/app/purchase/purchase-order-preview/purchase-order-preview.component.ts

```diff
@@ -0,0 +1,375 @@
+                        if (res.fields[key]) {
+                            this.formFields[res.fields[key].name] = [];
+                            this.formFields[res.fields[key].name] = res.fields[key];
+                        }
+                    });
+                }
+                if (this.formFields && this.formFields['taxName']) {
+                    this.shouldShowTrnGstField = true;
+                } else {
+                    this.shouldShowTrnGstField = false;
+                }
+            }
+        });
+    }
+
+    /**
+     * This will update the value of input variables on change
+     *
+     * @param {SimpleChanges} changes
+     * @memberof PurchaseOrderPreviewComponent
+     */
+    public ngOnChanges(changes: SimpleChanges): void {
+        if (changes.purchaseOrders && changes.purchaseOrders.currentValue && changes.purchaseOrders.currentValue.items) {
+            this.filteredData = changes.purchaseOrders.currentValue.items;
+        }
+
+        if (changes.purchaseOrderUniqueName && changes.purchaseOrderUniqueName.currentValue && changes.purchaseOrderUniqueName.currentValue !== this.purchaseOrder.uniqueName) {
+            this.purchaseOrderUniqueName = changes.purchaseOrderUniqueName.currentValue;
+            this.getPurchaseOrder();
+        }
+    }
+
+    /**
+     * This will get called after component has initialized
+     *
+     * @memberof PurchaseOrderPreviewComponent
+     */
+    public ngAfterViewInit(): void {
+        this.searchElement.nativeElement.focus();
+        fromEvent(this.searchElement.nativeElement, 'input')
+            .pipe(
+                debounceTime(500),
+                distinctUntilChanged(),
+                map((event: any) => event.target.value)
+            )
+            .subscribe((term => {
+                this.filteredData = this.purchaseOrders.items.filter(item => {
+                    return item.voucherNumber.toLowerCase().includes(term.toLowerCase()) ||
+                        item.vendor.name.toLowerCase().includes(term.toLowerCase()) ||
+                        item.voucherDate.includes(term) ||
+                        item.grandTotal.amountForAccount.toString().includes(term);
+                });
+            }))
+    }
+
+    /**
+     * This will get the purchase order details
+     *
+     * @returns {void}
+     * @memberof PurchaseOrderPreviewComponent
+     */
+    public getPurchaseOrder(): void {
+        if (this.isLoading) {
+            return;
+        }
+
+        this.isLoading = true;
+        let getRequest = { companyUniqueName: this.companyUniqueName, poUniqueName: this.purchaseOrderUniqueName };
+        this.purchaseOrderService.getPreview(getRequest).subscribe(response => {
+            this.isLoading = false;
+            if (response) {
+                if (response.status === "success") {
+                    this.purchaseOrder = response.body;
+
+                    if (this.purchaseOrder.account.billingDetails.country) {
```

Comment:
Consider adding nullity check on `account` and `billingDetails`

Reply:
it can't be null because i initialized it in constructor